core/src/main/java/org/apache/iceberg/util/Tasks.java

```diff
@@ -88,8 +89,17 @@ public UnrecoverableException(Throwable cause) {
     private long maxSleepTimeMs = 600000; // 10 minutes
     private long maxDurationMs = 600000;  // 10 minutes
     private double scaleFactor = 2.0;     // exponential
+    private boolean hasStarted = false;
```

Comment:
Since this can be updated in `runParallel`, should we consider making this an `AtomicBoolean` like the booleans defined inside of `runParallel` are?

Reply:
Looking closer, it seems like this wouldn't actually be set in some other thread, so I suppose it's ok. I would defer to your expertise on the issue 🙂

Reply:
`runParallel` starts threads, but i agree with you, the field isn't mutated in such a thread.